frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -454,6 +454,7 @@ export default ({
             beforeRequest
           }
         }).catch((e) => {
+          // TODO: add Retry button on the right side
```

Comment:
> I'm confused by this comment... isn't the point of this PR to implement this?

We make two calls to the backend when creating a new message. The first call is for `fetchServerTime` and second one is to publish event. When we create a message without any attachments, the pending message would be created after `fetchServerTime` is successfully executed, and before the event is actually published. If the user is offline, the first call (`fetchServerTime`) would be firstly failed, so no pending message would be created.
So, I thought it would be difficult to have the situation when users would have pending message which is failed to be published. So I ignored it here. But added TODO comment.

Reply:
@Silver-IT

1. If we are offline, the send button should be disabled, so that the user can't send any messages (please feel free to update this PR to do that).
2. If the message fails to send, the retry button should appear. This `.catch` doesn't do that. It needs to do that, as that is part of Issue #1890 - if the message fails to send for whatever reason, the retry button must appear

Reply:
Okay.